pkg/k8s/endpointsynchronizer/cep.go

```diff
@@ -108,7 +111,7 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 
 				// Serialize the endpoint into a model. It is compared with the one
 				// from before, only updating on changes.
-				mdl := e.GetCiliumEndpointStatus()
+				mdl := e.GetCiliumEndpointStatus(epSync.Allocator)
```

Comment:
Having the interface as part of the endpoint would avoid touching this file for example